src/main/webapp/src/app/recommendation/recommendation.component.spec.ts

```diff
@@ -21,4 +23,28 @@ describe('RecommendationComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should only display recommendation form in initial state', () => {
+
+    spyOn(component, 'onFormSubmit');
+    expect(component.onFormSubmit).not.toHaveBeenCalled();
+    expect(fixture.debugElement.query(By.css('app-recommendation-form'))).toBeDefined;
+    expect(fixture.debugElement.query(By.css('app-recommendation-list'))).toBeNull;
+    
+  });
+
+  it('should display recommendation form and list after form submission', () => {
+    expect(fixture.debugElement.query(By.css('app-recommendation-form'))).toBeTruthy;
+    expect(fixture.debugElement.query(By.css('app-recommendation-list'))).toBeFalsy;
+    
+    spyOn(component, 'onFormSubmit');
+    const hiddenGems: HiddenGem[] = [];
+    component.onFormSubmit(hiddenGems);
+    fixture.detectChanges();
+    expect(component.onFormSubmit).toHaveBeenCalled();
```

Comment:
I don't think you need the spy in this test, you're the one calling component.onFormSubmit() in the test.

Reply:
I removed the spy, I think it is needed to determine the expect(component.onFormSubmit).toHaveBeenCalled()? So I just checked the attributes the function should be setting when called (formSubmitted == True, HiddenGems = FormHiddenGemOutput?